unity-ggjj/Assets/Scripts/GameState/InvestigationState.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Ink.Runtime;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class InvestigationState : MonoBehaviour, IInvestigationState
+{
+    public enum ChoiceType
+    {
+        Talk,
+        Move
+    }
+
+    [SerializeField] private MenuOpener InvestigationMainMenuOpener;
+    [SerializeField] private ChoiceMenu InvestigationTalkMenu;
+    [SerializeField] private ChoiceMenu InvestigationMoveMenu;
+    
+    private readonly List<string> _examinedChoices = new();
+    private readonly List<string> _unlockedTalkChoices = new();
+    private readonly List<string> _unlockedMoveChoices = new();
+    private List<Choice> _moveOptions;
+    private List<Choice> _talkOptions;
+
+    public void UnlockChoice(string choice, ChoiceType type)
+    {
+        switch (type)
+        {
+            case ChoiceType.Talk:
+                _unlockedTalkChoices.Add(choice);
+                break;
+            case ChoiceType.Move:
+                _unlockedMoveChoices.Add(choice);
+                break;
+            default:
+                throw new ArgumentOutOfRangeException(nameof(type), type, null);
+        }
+    }
+
+    public bool IsChoiceUnlocked(string choice, ChoiceType type)
+    {
+        return type switch
+        {
+            ChoiceType.Talk => _unlockedTalkChoices.Contains(choice),
+            ChoiceType.Move => _unlockedMoveChoices.Contains(choice),
+            _ => false
+        };
+    }
+
+    // ReSharper disable Unity.PerformanceAnalysis
+    public void OpenWithChoices(List<Choice> talkOptions, List<Choice> moveOptions)
+    {
+        InvestigationMainMenuOpener.OpenMenu();
+        _talkOptions = talkOptions;
+        _moveOptions = moveOptions;
+    }
+
+    public void Clear()
+    {
+        _examinedChoices.Clear();
+    }
+
+    public void OpenTalkMenu()
+    {
+        InvestigationMainMenuOpener.CloseMenu();
+        InvestigationTalkMenu.Initialise(_talkOptions);
+    }
+
+    public void OpenMoveMenu()
+    {
+        InvestigationMainMenuOpener.CloseMenu();
+        InvestigationMoveMenu.transform.parent.gameObject.SetActive(true);
+        InvestigationMoveMenu.Initialise(_moveOptions);
+        var selectableAndLabel = InvestigationMoveMenu.transform.GetComponentsInChildren<MenuItem>().Select(menuItem => (menuItem, menuItem.GetComponentInChildren<TextMeshProUGUI>().text)).ToList();
+        foreach (var valueTuple in selectableAndLabel)
```

Comment:
If a tuple is necessary, I think this syntax is a little more consise. Saves writing lots of `valueTuple`s

```suggestion
        foreach (var (menuItem, text) in selectableAndLabel)
```